src/main/java/com/minecolonies/coremod/placementhandlers/DimensionFluidHandler.java

```diff
@@ -0,0 +1,76 @@
+package com.minecolonies.coremod.placementhandlers;
+
+import com.ldtteam.structurize.placement.handlers.placement.IPlacementHandler;
+import com.ldtteam.structurize.util.BlockUtils;
+import com.minecolonies.api.compatibility.candb.ChiselAndBitsCheck;
+import com.minecolonies.api.util.WorldUtil;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FlowingFluidBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.ldtteam.structurize.api.util.constant.Constants.UPDATE_FLAG;
+
+/**
+ * Makes lava in the nether free and water everywhere else.
+ */
+public class DimensionFluidHandler implements IPlacementHandler
+{
+    @Override
+    public boolean canHandle(@NotNull World world, @NotNull BlockPos pos, @NotNull BlockState blockState)
+    {
+        return blockState.getBlock() instanceof FlowingFluidBlock;
+    }
+
+    @Override
+    public List<ItemStack> getRequiredItems(
+      @NotNull World world,
+      @NotNull BlockPos pos,
+      @NotNull BlockState blockState,
+      @Nullable CompoundNBT tileEntityData,
+      boolean complete)
+    {
+        if (WorldUtil.isNetherType(world) && blockState.getBlock() == Blocks.LAVA)
```

Comment:
I am pretty sure there is ways to define this. Also this needs a todo to fix this later on in the lifecycle. Sind George will add a callback for that at least on the fluid to check for "Spreading" fluids

Reply:
I'll add the todo, there seems no better way.